Doc/library/typing.rst

```diff
@@ -767,6 +812,13 @@ These can be used as types in annotations and do not support ``[]``.
 
    .. versionadded:: 3.10
 
+   .. deprecated:: 3.12
+      :data:`TypeAlias` is deprecated in favor of the :keyword:`type` statement,
+      which creates instances of :class:`TypeAliasType`.
+      Note that while :data:`TypeAlias` and :class:`TypeAliasType` serve
+      similar purposes and have similar names, they are distinct and the
+      latter is not the type of the former.
```

Comment:
I'd like for us to also make clear that we have no planned removal date right now, similar to the language we use in the deprecation note for `typing.Text`: https://github.com/python/cpython/blob/0a7796052acb9cec8b13f8d0a5f304f56f26ec5b/Doc/library/typing.rst?plain=1#L2115-L2117